### DISCUSSION TITLE
URL path with colon (":") results in 404 not found

### DISCUSSION THREAD

**Discussion by dwinston** [encode/starlette#1673] [Potential Issue] [answered]
I find that I can't GET or POST to a path containing a colon (":"). An example in FastAPI (which uses Starlette):

```python
from fastapi import FastAPI

app = FastAPI()

@app.get("/items:queue")
def queue_new_item():
    return "An item has been queued for later inclusion."
```

On `uvicorn main:app --reload` (FYI Python 3.9.12 and FastAPI 0.78.0 on MacOS), I get a 404 response from `GET http://127.0.0.1:8000/items:queue`.

I also filed this issue in the fastapi repo (https://github.com/tiangolo/fastapi/issues/4984), but I suspect that Starlette's use of colons for path parameter converters (https://www.starlette.io/routing/) may be playing a role here.

Is this a bug? I am trying to follow the official google cloud APIs "custom methods" pattern <https://cloud.google.com/apis/design/custom_methods>:

>For custom methods, they should use the following generic HTTP mapping:
> https://service.name/v1/some/resource/name:customVerb

**Comment by dwinston** [comment]
```python
from starlette.routing import compile_path

compile_path("/queries:run")
```
outputs
```
(re.compile(r'^/queries$', re.UNICODE), '/queries:run', {})
```
This seems wrong!

**Comment by Kludex** [comment, marked as answer]
This was already fixed. I don't recall the version of the fix. :eyes:
